src/fromager/settings.py

```diff
@@ -82,6 +82,18 @@ def resolver_include_sdists(
             resolve_dist.get("include_sdists"), default
         )
 
+    def build_dir(self, pkg: str, sdist_root_dir: pathlib.Path) -> pathlib.Path:
+        p = self.get_package_settings(pkg)
+        if p.get("build_dir"):
+            input_build_dir = pathlib.Path(p.get("build_dir"))
+            # hack: make path absolute to ensure that any directory escaping is contained
+            if ".." in str(input_build_dir):
+                input_build_dir = input_build_dir.resolve()
```

Comment:
We could just raise an exception in this case instead of producing a value that's hard to predict.